test/distributions/test_distributions.py

```diff
@@ -4665,6 +4665,87 @@ def test_mean(self):
             else:
                 self.assertEqual(pytorch_dist.mean, scipy_dist.mean(), msg=pytorch_dist)
 
+    def _test_discrete_distribution_mode(self, dist, sanitized_mode):
+        # We cannot easily check the mode for discrete distributions, but we can look left and right
+        # to ensure the log probability is smaller than at the mode.
+        isfinite = dist.mode.isfinite().reshape(dist.batch_shape + (dist.event_shape.numel(),)).all(-1)
+        for step in [-1, 1]:
+            log_prob_mode = dist.log_prob(sanitized_mode)
+            if isinstance(dist, OneHotCategorical):
+                idx = (dist._categorical.mode + 1) % dist.probs.shape[-1]
+                other = torch.nn.functional.one_hot(idx, num_classes=dist.probs.shape[-1]).to(dist.mode)
+            else:
+                other = dist.mode + step
+            mask = isfinite & dist.support.check(other)
+            self.assertTrue(mask.any() or dist.mode.unique().numel() == 1)
+            other = torch.where(mask, other, dist.sample())
+            log_prob_other = dist.log_prob(other)
+            np.testing.assert_array_less(log_prob_other[mask], log_prob_mode[mask])
+
+    def _test_continuous_distribution_mode(self, dist, sanitized_mode):
```

Comment:
Nice test!  One idea to make the perturbation a bit more general and naturally multivariate would be to add small uniform noise in unconstrained space:
```py
num_points = 10
t = transform_to(dist.support)
unconstrained_mode = t.inv(mode)
perturbation = torch.rand((num_points,) + unconstrained_mode.shape).sub(0.5).mul(1e-5)
perturbed_mode = t(unconstrained_mode + perturbation)
assert (dist.log_prob(mode) >= dist.log_prob(perturbed_mode)).all()
```
WDYT?

Reply:
Much nicer!